src/providers/auth/auth.ts

```diff
@@ -58,12 +65,57 @@ export class AuthProvider {
   }
 
   /**
-   * signs a user out
+   * authenticates the user via cordova native plugin
+   * or fallsback to web popup authentication with firebase
    *
    * @returns {Observable<FirebaseUserProfile>}
    * @memberof AuthProvider
    */
-  signout(): Observable<FirebaseUserProfile> {
+  facebookAuth(): Observable<FirebaseUserProfile> {
+
+    const facebookPromise = this.platform.is('cordova') ?
+      this.nativeFacebookAuth() :
+      this.afAuth.auth.signInWithPopup(new firebase.auth.FacebookAuthProvider())
+        .then(response => response.user);
```

Comment:
Does this seem good in terms of readability?

Reply:
We'll let prettier decide 😂 😂